### PR TITLE
Feature/ipregistry.co

### PR DESCRIPTION
**Pull request by leotrubach**

### CODE REVIEW

File: modules/sfp_ipregistry.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+# -------------------------------------------------------------------------------
+
+import json
+import urllib.parse
+
+from spiderfoot import SpiderFootEvent, SpiderFootPlugin
+
+
+class sfp_ipregistry(SpiderFootPlugin):
+    __name__ = 'sfp_ipregistry'
```
Comment (smicallef) [inline modules/sfp_ipregistry.py:20]: Why is this needed?

File: modules/sfp_ipregistry.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    meta = {
+        "name": "IP Registry",
```
Comment (smicallef) [inline modules/sfp_ipregistry.py:22]: The name is technically `ipregistry` according to their website so I suggest we go with that. Please update this across the whole file.

File: modules/sfp_ipregistry.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            self.emit("PHYSICAL_COORDINATES", physical_location, pevent)
+
+    def generate_security_events(self, security, pevent):
+        if not isinstance(security, dict):
+            return
+        malicious = any(
+            security.get(k) for k in ("is_abuser", "is_attacker", "is_threat")
+        )
+        if malicious:
+            self.emit("MALICIOUS_IP_ADDRESS", pevent.data, pevent)
```
Comment (smicallef) [inline modules/sfp_ipregistry.py:141]: This should be formatted the same as other modules reporting maliciousness.

File: modules/sfp_ipregistry.py (hunk 1 above)
Comment (leotrubach) [reply modules/sfp_ipregistry.py:20]: If __name__ is not set the following code
```
self.sf.error(f"You enabled {self.__name__} but did not set an API key!", False)
```
outputs ugly message.

Comment (bcoles) [reply modules/sfp_ipregistry.py]: Try this.

```python
        if self.opts["api_key"] == "":
            self.sf.error(
                f"You enabled {self.__class__.__name__} but did not set an API key!",
                False,
            )
            self.errorState = True
            return None
```

Comment (bcoles) [reply modules/sfp_ipregistry.py]: Alternatively, hardcode the module name.

```python
        if self.opts["api_key"] == "":
            self.sf.error(
                "You enabled sfp_ipregistry but did not set an API key!",
                False,
            )
            self.errorState = True
            return None
```
